code/modules/research/experimentor.dm

```diff
@@ -1,98 +1,55 @@
 //this is designed to replace the destructive analyzer
 
-//NEEDS MAJOR CODE CLEANUP
+GLOBAL_LIST_INIT(critical_items,typecacheof(list(/obj/item/construction/rcd,/obj/item/grenade,/obj/item/aicard,/obj/item/storage/backpack/holding,/obj/item/slime_extract,/obj/item/onetankbomb,/obj/item/transfer_valve)))
+
+/datum/experiment_type
+	var/name = "Adminhelp"
+	var/hidden = FALSE //Whether we should hide this from newly made experimentors
+	var/list/experiments = list()
+	var/list/item_results = list()
+
+/datum/experiment_type/proc/get_valid_experiments(obj/machinery/rnd/experimentor/E,obj/item/O,bad_things_coeff,only_bad=FALSE)
+	var/list/weighted_experiments = list()
+	for(var/datum/experiment/EX in experiments)
+		if(!EX.weight || !EX.can_perform(E,O))
+			continue
+		if(EX.is_bad)
+			weighted_experiments[EX] = EX.weight / bad_things_coeff

```

Comment:
hm. can you run the calculations on this with your part improvements? I'm getting a bad_thing_coeff of roughly 80 with gooftier, which would lower the weight of bad clone and vaporize to about 10 each. (800 / 80 = 10)
I'm thinking it looks similar with lower tier parts. Good idea to divide, though. Just the factor should be capped/lower methinks.